packages/sign_in_with_apple/sign_in_with_apple_web/lib/sign_in_with_apple_web.dart

```diff
@@ -0,0 +1,133 @@
+  /// Registers this class as the default instance of [SignInWithApplePlatform].
+  static void registerWith(Registrar registrar) {
+    SignInWithApplePlatform.instance = SignInWithApplePlugin();
+  }
+
+  @override
+  Future<bool> isAvailable() async {
+    return true;
+  }
+
+  @override
+  Future<AuthorizationCredentialAppleID> getAppleIDCredential({
+    required List<AppleIDAuthorizationScopes> scopes,
+    WebAuthenticationOptions? webAuthenticationOptions,
+    String? nonce,
+    String? state,
+  }) async {
+    try {
+      final options = SignInWithAppleInitOptions(
+        clientId: webAuthenticationOptions!.clientId,
+        redirectURI: webAuthenticationOptions.redirectUri.toString(),
+        scope: [
+          for (final scope in scopes)
+            if (scope == AppleIDAuthorizationScopes.email)
+              'email'
+            else if (scope == AppleIDAuthorizationScopes.fullName)
+              'name',
+        ].join(' '),
+        state: state,
+        nonce: nonce,
+        usePopup: true,
+      );
+
+      init(options);
+      final response = await promiseToFuture<SignInResponseI>(signIn());
+
+      return AuthorizationCredentialAppleID(
+        authorizationCode: response.authorization.code,
+        identityToken: response.authorization.id_token,
+        state: response.authorization.state,
+        email: response.user?.email,
+        givenName: response.user?.name?.firstName,
+        familyName: response.user?.name?.lastName,
+        userIdentifier: null,
+      );
+    } catch (e) {
+      // error per https://developer.apple.com/documentation/sign_in_with_apple/signinerrori
+      final errorProp = getProperty(e, 'error');
+      final errorCode = errorProp is String ? errorProp : 'UNKNOWN_SIWA_ERROR';
+
+      throw SignInWithAppleCredentialsException(
+        message: 'Authentication failed with $errorCode',
+      );
+    }
+  }
+}
+
+@JS()
+@anonymous
+class SignInWithAppleInitOptions {
+  external String? get clientId;
+  external String? get scope;
+  external String? get redirectURI;
+  external String? get state;
+  external String? get nonce;
+  external bool? get usePopup;
+
+  // Must have an unnamed factory constructor with named arguments.
+  external factory SignInWithAppleInitOptions({
+    String? clientId,
+    String? scope,
+    String? redirectURI,
+    String? state,
+    String? nonce,
+    bool? usePopup,
+  });
+}
+
+@JS('console.log')
+external void log(Object o);
+
+@JS('AppleID.auth.init')
+external void init(SignInWithAppleInitOptions options);
+
+@JS('AppleID.auth.signIn')
+external Object /* like Future<SignInResponseI> */ signIn();
+
+/// Sign in with Apple authorization response
+///
+/// Spec: https://developer.apple.com/documentation/sign_in_with_apple/signinresponsei
+@JS()
+@anonymous
+class SignInResponseI {
+  external AuthorizationI get authorization;
+  external UserI? get user;
+}
+
+@JS()
+@anonymous
```

Comment:
@tp the version where we don't depend on the SiwA JS script didn't work?

Reply:
> @tp the version where we don't depend on the SiwA JS script didn't work?

See the opening comment for details, but in the end not using the JS would impose more implementation effort on each consuming app. Hence I choose this as the default, but the other approach would be a valid addition/refinement.

If we don't use Apple JS, it'd mean we'd have to either redirect ourselves in the current frame (which is a bit bad as we loose the place in the app / have to restart it), or have to open "our own" pop-up, but then the final redirect has to be to a new page which is able communicate with the host app (this is then something the consuming app would have to implement).

I would prefer the "own pop-up" (or `iframe`, if Apple allows it), but didn't want to make it too complicated for consumers to implement.